src/earn/EarnConfirmationScreen.tsx

```diff
@@ -74,20 +78,50 @@ export default function EarnConfirmationScreen({ route }: Props) {
     [withdrawToken, pool.pricePerShare, inputAmount]
   )
 
-  // Will need this to handle preparing a claim transaction, a withdrawal transaction and a withdrawal and claim transaction
   const {
-    result: prepareTransactionsResult,
-    loading: isPreparingTransactions,
-    error: prepareTransactionError,
-  } = usePrepareWithdrawAndClaimTransactions({
-    amount: new BigNumber(withdrawAmountInDepositToken).dividedBy(pool.pricePerShare[0]).toString(),
-    pool,
-    walletAddress,
-    feeCurrencies,
-    hooksApiUrl,
-    rewardsPositions,
-    useMax,
-  })
+    // @ts-expect-error prepareTransactionsResult & swapTransaction are only set when mode === 'deposit' or 'swap-deposit'
+    prepareTransactionsResult: { prepareTransactionsResult, swapTransaction } = {},
+    refreshPreparedTransactions,
+    clearPreparedTransactions,
+    prepareTransactionError,
+    isPreparingTransactions,
+  } = usePrepareTransactions(mode)
+
+  const handleRefreshPreparedTransactions = (
+    amount: BigNumber,
+    token: TokenBalance,
+    feeCurrencies: TokenBalance[]
+  ) => {
+    if (!walletAddress || !isAddress(walletAddress)) {
+      Logger.error(TAG, 'Wallet address not set. Cannot refresh prepared transactions.')
+      return
+    }
+
+    return refreshPreparedTransactions({
+      amount: new BigNumber(amount).dividedBy(pool.pricePerShare[0]).toString(),
+      token,
+      walletAddress,
+      feeCurrencies,
+      pool,
+      hooksApiUrl,
+      // Mode === shortcutId, except for exit which is withdraw
+      shortcutId: mode === 'exit' ? 'withdraw' : mode,
```

Comment:
So exit maps to 'prepareWithdrawAndClaimTransactions' which doesn't appear to have shortcutId as a parameter. So why is this needed?

Reply:
Fixed in 98a8bd00d6a0d1ecef892ad8d7bdc34d39e86b63.